README.md

```diff
@@ -1,2 +1,103 @@
+The ideal index policy for the target container would look like this:
+
+```xml
+{
+    "indexingMode": "consistent",
+    "automatic": true,
+    "includedPaths": [
+        
+    ],
+    "excludedPaths": [
+        {
+            "path": "/*"
+        },
+        {
+            "path": "/\"_etag\"/?"
+        }
+    ]
+}
+```
+
+2) Create the Job contianer to use for tracking status etc. with metadata documents. This container has to use a partition key definition of `/pk` - and the standard indexing policy should be used.
+
+3) Sepcify the environment variables according to the (Configuration)[#Configuration] section above.
+
+#### Building the application
+You can use `mvn install` to build the jar file. The application will produce a fat-jar (one single jar file containing all dependencies). This file will be located in the traget folder and have a name similar to `azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar`.
+
+#### Creating a new ingestion job
+Each ingestion will be tracked in a `Job` - the job and it's batches (fragments of input files) is tracked in a Cosmos DB Container used to store the metadata. The workers will then load-balance the ingestion across these batches.
+
+To create a new ingestion job - run the following command line
+```
+java -cp ./azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar com.azure.cosmos.samples.distributedbulk.Main create SampleJob01 ^.*\.json$
+```
+
+A slightly different syntax can be used to create the job quicker when you know that all your input files have the same number of lines/documents. In this case not each file is downloaded form Storage to count the target number of lines - but this step is done only for one file and then the same numbe rof lines is used for all files.
+```
+java -cp ./azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar com.azure.cosmos.samples.distributedbulk.Main createUnifrom SampleJob01 ^.*\.json$
+```
+
+Instead of `SampleJob01`use a unique identifier for your job. The `^.*\.json$` regex is used to identify the search pattern to find the files to be ingested (in the Storage account/container defined via the environment variables)
+
+** NOTE: ** please only execute the command to create a job from a single worker
+
+
+#### Ingesting the data
+This step can be executed from multiple workes. On each worker execute the following command.
+
+```
+java -cp ./azure-cosmos-distributed-bulk-sample-1.0-SNAPSHOT-jar-with-dependencies.jar com.azure.cosmos.samples.distributedbulk.Main process SampleJob01
+```
+
+Instead of `SampleJob01`use the identifier for your job you created.
+
+#### Cleaning up status for a job
+Just in case you want to re-reun ingestion for a job with the same name/identifier, it is possible to also delete all metadata documents for a job.
```

Comment:
```suggestion
Just in case you want to re-run ingestion for a job with the same name/identifier, it is possible to also delete all metadata documents for a job.
```